phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java

```diff
@@ -743,6 +764,32 @@ public int run(String[] args) throws Exception {
         }
     }
 
+    private void deleteBeforeRebuild(Connection conn) throws SQLException, IOException {
+        if (MetaDataUtil.isViewIndex(pIndexTable.getPhysicalName().getString())) {
+            throw new IllegalArgumentException(String.format(
+                    " %s is a view index table. Delete all before rebuild is not supported for view indexes",
+                    indexTable));
+        }
+
+        if (isLocalIndexBuild) {
+            throw new IllegalArgumentException(String.format(
+                    " %s is a local index.  Delete all before rebuild is not supported for local indexes", indexTable));
+        } else {
+            Admin admin = null;
+            ConnectionQueryServices queryServices = conn.unwrap(PhoenixConnection.class).getQueryServices();
+            try {
+                admin = queryServices.getAdmin();
+                TableName tableName = TableName.valueOf(qIndexTable);
+                admin.disableTable(tableName);
+                admin.truncateTable(tableName, false);
+            } finally {
+                if (admin != null) {
+                    admin.close();
```

Comment:
I've seen C# using being useful in such cases. Did some search and it seems like in Java 7+ onwards we can use try -with-resources. The code would look like:
try( Admin admin = queryServices.getAdmin()) {
                TableName tableName = TableName.valueOf(qIndexTable);
                admin.disableTable(tableName);
                admin.truncateTable(tableName, false);
}

No need to add finally block. Makes it smaller and easier to read.

Reply:
Good catch! Yes, it is a good practice to use try-with-resources in java as well.

Reply:
Sure, but code convention seems to be try/finally throughout this codebase.